instance/tests/api/test_openedx_instance.py

```diff
@@ -98,6 +98,48 @@ def assert_active_appserver(self, instance_id, appserver_id):
             )
         return response
 
+    def test_instance_list_efficiency(self):
+        """
+        The number of database queries required to fetch /api/v1/instance/
```

Comment:
@bradenmacdonald Again, I'm just being pedantic here, so I agree this is one way to test it. My point is kind of two fold:

* I'd rather say it asserts that no matter how many instances there are, querying for list of instances takes 9 queries.
* How much queries certain code path performs requires a bit too deep knowledge about implementation than test should have. (in my opinion)

> modify this test to detect queries_per_api_call automatically on the first iteration and then ensure it doesn't change on the 2-3 iterations of the loop

Might actually be a better way.